src/data/markdown/docs/01 guides/02 Using k6/14 Executors.md

```diff
@@ -0,0 +1,635 @@
+of time. This executor is equivalent to the global `vus` and `duration` options.
+
+| Option     | Type    | Description                                     | Default |
+|------------|---------|-------------------------------------------------|---------|
+| `vus`      | integer | Number of VUs to run concurrently.              | `1`     |
+| `duration` | string  | Total test duration (excluding `gracefulStop`). | -       |
+
+#### Examples
+
+- Run a constant 10 VUs for 10 seconds:
+
+<div class="code-group" data-props='{"labels": [ "constant-vus.js" ], "lineNumbers": "[true]"}'>
+
+```js
+import http from 'k6/http';
+
+export let options = {
+  discardResponseBodies: true,
+  scenarios: {
+    contacts: {
+      executor: 'constant-vus',
+      vus: 10,
+      duration: '10s',
+    }
+  }
+};
+
+export default function() {
+  http.get('https://test.k6.io/contacts.php');
+}
+```
+
+</div>
+
+
+### Ramping VUs
+
+A variable number of VUs execute as many iterations as possible for a specified
+amount of time. This executor is equivalent to the global `stages` option.
+
+| Option             | Type    | Description                                                                   | Default |
+|--------------------|---------|-------------------------------------------------------------------------------|---------|
+| `startVUs`         | integer | Number of VUs to run at test start.                                           | `1`     |
+| `stages`           | array   | Array of objects that specify the target number of VUs to ramp up or down to. | `[]`    |
+| `gracefulRampDown` | string  | Time to wait for iterations to finish before starting new VUs.                | `"30s"` |
+
+#### Examples
+
+- Run a two-stage test, ramping up from 0 to 100 VUs for 5 seconds, and down to 0 VUs
+  for 5 seconds:
+
+<div class="code-group" data-props='{"labels": [ "ramping-vus.js" ], "lineNumbers": "[true]"}'>
+
+```js
+import http from 'k6/http';
+
+export let options = {
+  discardResponseBodies: true,
+  scenarios: {
+    contacts: {
+      executor: 'ramping-vus',
+      startVUs: 0,
+      stages: [
+        { duration: '5s', target: 100 },
+        { duration: '5s', target: 0 },
+      ],
+      gracefulRampDown: '0s',
+    }
+  }
+};
+
+export default function() {
+  http.get('https://test.k6.io/contacts.php');
+}
+```
+
+</div>
+
+Note the setting of `gracefulRampDown` to 0 seconds, which could cause some
+iterations to be interrupted during the ramp down stage.
+
+
+### Constant arrival rate
+
+A fixed number of iterations are executed in a specified period of time.
+Since iteration execution time can vary because of test logic or the
+system-under-test responding more slowly, this executor will try to compensate
+by running a variable number of VUs--including initializing more in the middle
+of the test--in order to meet the configured iteration rate. This approach is
+useful for a more accurate representation of RPS, for example.
```

Comment:
It was supposed to be an em dash, to represent an interjection of sorts, and some Markdown engines render it properly, but not Gatsby apparently. So I'll change it to the HTML codes, unless you think we should reword the sentence to avoid it.

Reply:
Resolved in 2d07d72?

Reply:
> It was supposed to be an em dash,
I didn't know this :D, but yeah if it doesn't work in Gatsby ...